elife/mysql-client.sls

```diff
@@ -6,3 +6,16 @@ mysql-package-list:
             - mysql-client
             - python-mysqldb # py2, Salt 2017.7.x
             - python3-mysqldb # py3, Salt 2018.3+
+
+    # on Ubuntu 16.04, in Salt 2018.3, using Python3, there is a problem passing a 
+    # bytestring as an argument to a function that expects a regular string.
+    # this patch decodes any bytestring arguments into a regular string
+    file.patch:
+        - name: /usr/lib/python3/dist-packages/MySQLdb/cursors.py
+        - source: salt://elife/config/usr-lib-python3-dist-packages-MySQLdb-cursors.py.patch
+        # md5 of original file: aca2fb480a76e30f65601018916e86bd
+        # md5 of patched file:
+        - hash: 15d883d28d4548374522fad16eed560e
```

Comment:
isn't this likely to change if the file is changed upstream in a way that is still compatible with the patch?

Reply:
Then the hash check will fail and we'll know it's been fixed or the patch needs updating

It's less likely the `python3-mysqldb` package will change in 16.04 than the Salt module code.

I don't like patching these files but I don't see any movement in the ticket I opened with the Salt guys.